Notification server getting notifications out of order

This happens infrequently in end to end tests. The gap is always one on the handful of samples I have observed.

```
2019-12-20T09:11:47.6334279Z 35: Traceback (most recent call last):
2019-12-20T09:11:47.6334494Z 35:   File "/mnt/build/1/s/tests/receipts.py", line 73, in <module>
2019-12-20T09:11:47.6334604Z 35:     run(args)
2019-12-20T09:11:47.6334667Z 35:   File "/mnt/build/1/s/tests/receipts.py", line 59, in run
2019-12-20T09:11:47.6334741Z 35:     test(network, args, notifications_queue)
2019-12-20T09:11:47.6334817Z 35:   File "/mnt/build/1/s/tests/suite/test_requirements.py", line 38, in wrapper
2019-12-20T09:11:47.6334890Z 35:     return func(network, args, *nargs, **kwargs)
2019-12-20T09:11:47.6334967Z 35:   File "/mnt/build/1/s/tests/suite/test_requirements.py", line 38, in wrapper
2019-12-20T09:11:47.6335037Z 35:     return func(network, args, *nargs, **kwargs)
2019-12-20T09:11:47.6335109Z 35:   File "/mnt/build/1/s/tests/receipts.py", line 34, in test
2019-12-20T09:11:47.6408993Z 35:     check_commit(c.rpc("LOG_record", {"id": 42, "msg": msg}), result=True)
2019-12-20T09:11:47.6409505Z 35:   File "/mnt/build/1/s/tests/infra/checker.py", line 72, in __call__
2019-12-20T09:11:47.6409627Z 35:     ), f"Received notification of commit {n} after commit {self.notified_commit}"
2019-12-20T09:11:47.6409942Z 35: AssertionError: Received notification of commit 13 after commit 14
```

Noting another instance of this. We could log notification bodies as we receive them, but I don't know if this will tell us anything useful.

https://dev.azure.com/MSRC-CCF/CCF/_build/results?buildId=3863&view=logs&j=6df99028-3142-569d-f31c-c98e679660f8&t=25a1771e-fd5c-5beb-59a5-e7c11c122c81&l=17481

```
39: Traceback (most recent call last):
39:   File "/mnt/build/1/s/tests/e2e_logging.py", line 137, in <module>
39:     run(args)
39:   File "/mnt/build/1/s/tests/e2e_logging.py", line 122, in run
39:     network = test(network, args, notifications_queue)
39:   File "/mnt/build/1/s/tests/suite/test_requirements.py", line 38, in wrapper
39:     return func(network, args, *nargs, **kwargs)
39:   File "/mnt/build/1/s/tests/suite/test_requirements.py", line 38, in wrapper
39:     return func(network, args, *nargs, **kwargs)
39:   File "/mnt/build/1/s/tests/e2e_logging.py", line 74, in test
39:     check_commit(c.rpc("LOG_record", {"id": 42, "msg": msg}), result=True)
39:   File "/mnt/build/1/s/tests/infra/checker.py", line 72, in __call__
39:     ), f"Received notification of commit {n} after commit {self.notified_commit}"
39: AssertionError: Received notification of commit 10 after commit 11
30/34 Test #39: end_to_end_logging ...............***Failed   10.30 sec
```

--- a/tests/infra/notification.py
+++ b/tests/infra/notification.py
@@ -33,7 +33,7 @@ class PostQueueRequestHandler(http.server.BaseHTTPRequestHandler):
         pass
 
 
-class ThreadingPostQueueServer(http.server.ThreadingHTTPServer):
+class PostQueueServer(http.server.HTTPServer):
     def __init__(self, server_address, RequestHandlerClass, checker=None):
         assert (
             RequestHandlerClass is PostQueueRequestHandler
@@ -41,7 +41,7 @@ class ThreadingPostQueueServer(http.server.ThreadingHTTPServer):
         self.queue = queue.Queue()
         self.error_queue = queue.Queue()
         self.checker = checker
-        super(ThreadingPostQueueServer, self).__init__(
+        super(PostQueueServer, self).__init__(
             server_address, PostQueueRequestHandler
         )
 
@@ -54,21 +54,18 @@ class ThreadingPostQueueServer(http.server.ThreadingHTTPServer):
 
 @contextmanager
 def notification_server(server_info, checker=None):
-
     host = None
     port = []
     if server_info is not None:
         host, *port = server_info.split(":")
-
         if not host or not (port and port[0]):
             raise ValueError("Notification server host:port configuration is invalid")
     else:
         raise ValueError("Notification server host:port configuration is invalid")
 
-    with ThreadingPostQueueServer(
+    with PostQueueServer(
         (host, int(port[0])), PostQueueRequestHandler, checker
     ) as server:
-
         server_thread = threading.Thread(target=server.serve_forever)
         server_thread.daemon = True
         server_thread.start()

--- a/tests/infra/notification.py
+++ b/tests/infra/notification.py
@@ -41,9 +41,7 @@ class PostQueueServer(http.server.HTTPServer):
         self.queue = queue.Queue()
         self.error_queue = queue.Queue()
         self.checker = checker
-        super(PostQueueServer, self).__init__(
-            server_address, PostQueueRequestHandler
-        )
+        super(PostQueueServer, self).__init__(server_address, PostQueueRequestHandler)
 
     def get_queue(self):
         return self.queue

